src/System.Diagnostics.Tracing.Logger/src/System/Diagnostics/Tracing/Logger.cs

```diff
@@ -0,0 +1,206 @@
+                        return true;
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Log a message.   A message has a name, level (verbosity) and optional data (arguments).  The arguments 
+        /// is typically a anonymous type with as many fields as needed by the message.   The expectation is that
+        /// you can process the result programatically (by subscribing to it), but you can also render the message
+        /// as a string of the from  MessageName Level=level FieldName1=Value1  FieldName2=Value2 ...
+        /// 
+        /// Because calling this method typically requires a anonymous type to be generated for the 'arguments'
+        /// value if this logging happens in a 'hot' code path you should have a 'IsEnabled' check before calling
+        /// it to avoid having to make the object just to throw it away if logging is off.   
+        /// </summary>
+        /// <param name="messageName">The name for this logging message.  (e.g. ExceptionOccured, or DataItemRetrieved)</param>
+        /// <param name="level">The verbosity of this message.</param>
+        /// <param name="arguments">Any additional data.  Typically this is an anoymous type object.</param>
+        public virtual void Log(string messageName, LogLevel level, object arguments = null)
+        {
+            if (IsEnabled(level))
+            {
+                 Write(messageName, new LoggerArguments { Level = level, LoggerName = Name, Arguments = arguments });
+            }
+        }
+
+        /// <summary>
+        /// If the logger is enabled for 'level' then geneate a 'activityName'.Start message with the given argument.
+        /// It returns an IDisposable that when Disposed() will send a 'activityName'.Stop message with the SAME arguments 
+        /// object.  Thus the arguments object itself can be used as an ID that links the two together.
+        /// </summary>
+        public virtual IDisposable ActivityStart(string activityName, LogLevel level = LogLevel.Critical, object arguments = null)
+        {
+            var activity = new LoggerActivity(this, level, activityName, arguments);
+            if (IsEnabled(level))
+                Log(activityName + ".Start", level, arguments);
+            else
+                activity.Disposed = true;
+            return activity;
+        }
+
+        /// <summary>
+        /// Subscribe to the Logger, sending events to 'observer' for any logging that is enabled for verbosity 'level'.  
+        /// 
+        /// Note that in multi-subscriber cases you MAY get messages that have a more verbose level than you asked for.
+        /// You can filter them out yourself.   If there is only one subscriber (or all subscribers subscribe at the 
+        /// same verbosity.  It works fine.   
+        /// </summary>
+        public virtual IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Func<string, LogLevel, bool> filter)
```

Comment:
Is this the replacement for the LoggerProvider? I see the comment about using `DiagnosticListener.AllListeners` to get at a particular instance. I assume there's going to be an example of that somewhere else?